controllers/bootstrap/init.go

```diff
@@ -466,6 +516,43 @@ func (b *Bootstrap) CreateOrUpdateFromYaml(yamlContent []byte, alwaysUpdate ...b
 	return errMsg
 }
 
+// DeleteFromYaml takes [objectTemplate, b.CSData] and delete the object according to the objectTemplate
+func (b *Bootstrap) DeleteFromYaml(objectTemplate string, data interface{}) error {
+	var buffer bytes.Buffer
+	t := template.Must(template.New("newTemplate").Parse(objectTemplate))
+	if err := t.Execute(&buffer, data); err != nil {
+		return err
+	}
+
+	yamlContent := buffer.Bytes()
+	objects, err := util.YamlToObjects(yamlContent)
+	if err != nil {
+		return err
+	}
+
+	var errMsg error
+
+	for _, obj := range objects {
+		gvk := obj.GetObjectKind().GroupVersionKind()
+
+		_, err := b.GetObject(obj)
+		if errors.IsNotFound(err) {
+			klog.Infof("Not Found name: %s, namespace: %s, kind: %s, apiversion: %s/%s\n, skipping", obj.GetName(), obj.GetNamespace(), gvk.Kind, gvk.Group, gvk.Version)
+			continue
+		} else if err != nil {
+			errMsg = err
+			continue
+		}
+
+		klog.Infof("Deleting resource with name: %s, namespace: %s, kind: %s, apiversion: %s/%s\n", obj.GetName(), obj.GetNamespace(), gvk.Kind, gvk.Group, gvk.Version)
+		if err := b.DeleteObject(obj); err != nil {
+			errMsg = err
+		}
```

Comment:
Could add a wait function to make sure the object is deleted?

Reply:
sure, should I create a wait function to check whether the object is deleted? something similar to [WaitResourceReady](https://github.com/IBM/ibm-common-service-operator/blob/c870a4fd53f75a36f910d0a077ad68251771167f/controllers/bootstrap/init.go#L584:21)

Reply:
maybe we don't need wait function since no following operations need these deletions has to be done

Reply:
We need to make sure CRs are deleted before deleting the operator.